src/Native/configure.cmake

```diff
@@ -163,6 +163,17 @@ check_struct_has_member(
     "sys/select.h"
     HAVE_PRIVATE_FDS_BITS)
 
+check_cxx_source_compiles(
+    "
+    #include <sys/sendfile.h>
+    int main() { int i = sendfile(0, 0, 0, 0); }
+    "
+    HAVE_SENDFILE)
+
+check_function_exists(
+    fcopyfile
```

Comment:
From my experiences with with cmake feature checks, check_function_exists just generates a code file that tries using the specified function and then compiles the file. If the code file compiles, then it decides the function exists.
But if function can't be found without adding an include for the header file, the compile fails.

I figured since you needed to add an include in the actual code file, you would probably need to add an include in the feature check.

You can look at the bin/obj/OSX.x64.Debug/Native/Common/pal_config.h that gets generated to check that the feature check is working correctly.